src/component/legend/LegendModel.ts

```diff
@@ -362,6 +362,13 @@ class LegendModel<Ops extends LegendOption = LegendOption> extends ComponentMode
                     name: dataItem as string
                 };
             }
+
+            const isNil = (val: any) => val === null || val === undefined || val === '';
```

Comment:
Although not suggest, I think it may be valid for a series name to be `''`, so `isNil` can be replaced by simple checking `== null`.

Reply:
yeh, although the series name `''` is useless for legend, it is indeed a valid value. I changed the code and removed the handling of `''`